opentracing-api/src/main/java/io/opentracing/ActiveSpan.java

```diff
@@ -106,4 +111,13 @@
         ActiveSpan activate();
     }
 
+    /**
+     * The {@link ActiveSpan.Observer} interface learns of capture, activate, and deactivate calls for a given
+     * {@link ActiveSpan} instance and any of its "descendants" from a capture/activate standpoint.
+     */
+    interface Observer {
+        void onCapture(ActiveSpan captured, Continuation destination);
+        void onActivate(Continuation source, ActiveSpan justActivated);
+        void onDeactivate(ActiveSpan activeSpan, Finishable finisher);
```

Comment:
by factoring out the `Finishable` interface from `Span` proper, there's no longer a need to have `ActiveSpan` inherit from `Span` rather than `BaseSpan`... we're also back to a world where you can't call `.finish()` on an `ActiveSpan`, and IMO that's a good thing.

I am uneasy about the precise signatures for this `Observer` interface... it could be smaller/simpler and still satisfy the requirements here (to factor out refcounting).